File: Frontend/src/app/viewModels/requests/EvaluateRouteVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model class for an Evaluate Route request
 */
export class EvaluateRouteVM implements toJSONString {

    /**
     * Construct's an instance of {@link EvaluateRouteVM}
     * 
     * @param evaluation given evaluation
     */
    constructor(
        public evaluation: Number
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify({ "evaluation": this.evaluation })
    }

}

File: Backend/Data/ViewModels/Request/EvaluateRouteVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View model class for evaluate route request
    /// </summary>
    public class EvaluateRouteVM
    {
        public long Evaluation { get; set; }

        /// <summary>
        /// Class's constructor method
        /// </summary>
        /// <param name="evaluation">Evaluation</param>
        public EvaluateRouteVM(long evaluation)
        {
            this.Evaluation = evaluation;

        }
    }
}